rpc/backend/utils.go

```diff
@@ -107,7 +107,10 @@ func (b *Backend) getAccountNonce(
 				break
 			}
 
-			sender := ethMsg.GetSender()
+			sender, err := ethMsg.GetSender(b.chainID)
+			if err != nil {
+				b.logger.Debug("failed to parse from field", "hash", ethMsg.Hash, "error", err.Error())
+			}
 			if sender == accAddr {
```

Comment:
Wouldn't sender be undefined/unpredictable if err != nil here? If it is fine not incrementing the nonce in this case, then it looks good to me but maybe returning nonce directly above is cleaner